homeassistant/components/qnap/sensor.py

```diff
@@ -208,11 +203,8 @@ def setup_platform(
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the QNAP NAS sensor."""
-    api = QNAPStatsAPI(config)
-    api.update()
-
-    # QNAP is not available
-    if not api.data:
+    coordinator = QnapCoordinator(hass, config)
+    if not coordinator:
```

Comment:
I think the coordinator will never fail to get created.
What you need to check is the "initial" update:
```suggestion
    await coordinator.async_refresh()
    if not coordinator.last_update_success:
```

Reply:
Note: you'll need to convert the function to `async_setup_platform` for this to work